### PR TITLE
added fit wizard to muon feature flags

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**

**To test:**

Open muon analysis
Load some data
Go to the fitting tab
There will not be an button called "fit scrip generator"
Close mantid

Open mantid.user.properties
Make sure you have `muon.GUI = fit_wizard:1`
Open muon analysis
Load some data
Go to the fitting tab
There will be an button called "fit scrip generator"
Check that it works

*There is no associated issue.*

This is not in the release notes as its not fully exposed to users.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by robertapplin — CHANGES_REQUESTED** [review]
Looks good, I haven't tested it yet but will hopefully do that tomorrow.

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/features/add_fitting.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
```
[the remark this answers is not in the document]
Comment (AnthonyLim23) [reply qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/features/add_fitting.py:3]:
```suggestion
# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
```

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/features/add_fitting.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantidqtinterfaces.Muon.GUI.Common.features.add_feature import AddFeature
+
+
+FITWIZARD = "fit_wizard"
```
[the remark this answers is not in the document]
Comment (AnthonyLim23) [reply qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/features/add_fitting.py:10]: I have gone for fit wizard, as that feels like a more user friendly name. There is some intuition that a "wizard" makes the thing (in this case fits) easier.

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/fitting_tab_widget/fitting_tab_widget.py [hunk 3]
```diff
@@ -36,3 +36,6 @@ def __init__(self, context, parent):
 
         context.update_view_from_model_notifier.add_subscriber(
             self.fitting_tab_presenter.update_view_from_model_observer)
+
+    def show_fit_script_generator(self)->None:
+        self.fitting_tab_view.show_fit_script_generator()
```
Comment (AnthonyLim23) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/fitting_tab_widget/fitting_tab_widget.py:41]:
```suggestion
        """
        Show the fit script generator in the fitting interface
        """
        self.fitting_tab_view.show_fit_script_generator()
```

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/fitting_widgets/basic_fitting/basic_fitting_view.py [hunk 4]
```diff
@@ -322,3 +322,6 @@ def disable_view(self) -> None:
     def enable_view(self) -> None:
         """Enable all widgets in this fitting widget."""
         self.setEnabled(self.workspace_selector.number_of_datasets() != 0)
+
+    def show_fit_script_generator(self) ->None:
+        self.fit_controls.show_fit_script_generator()
```
Comment (AnthonyLim23) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/fitting_widgets/basic_fitting/basic_fitting_view.py:327]:
```suggestion
        """
        Show the fit script generator in the fitting interface
        """
        self.fit_controls.show_fit_script_generator()
```

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/fitting_widgets/basic_fitting/fit_controls_view.py [hunk 5]
```diff
@@ -25,6 +25,9 @@ def __init__(self, parent: QWidget = None):
         # Comment out this line to show the 'Fit Generator' button
         self.fit_generator_button.hide()
 
+    def show_fit_script_generator(self):
```
Comment (AnthonyLim23) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/fitting_widgets/basic_fitting/fit_controls_view.py:28]:
```suggestion
    def show_fit_script_generator(self) -> None:
```

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/features/add_fitting.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (robertapplin) [reply qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/features/add_fitting.py:10]: To me its hard to understand what a 'fit wizard' does, whereas with 'fit script generator' I find it is much clearer what its purpose is

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/fitting_widgets/basic_fitting/fit_controls_view.py [hunk 6]
```diff
@@ -25,6 +25,9 @@ def __init__(self, parent: QWidget = None):
         # Comment out this line to show the 'Fit Generator' button
         self.fit_generator_button.hide()
 
+    def show_fit_script_generator(self) -> None:
+        self.fit_generator_button.show()
```
Comment (robertapplin) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/fitting_widgets/basic_fitting/fit_controls_view.py:29]: Can you add a function comment here too

**Review by robertapplin — APPROVED** [review]
Ready to be shipped

**Pull request merged**